keras_cv/models/classification/image_classifier.py

```diff
@@ -0,0 +1,189 @@
+# Copyright 2023 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Image classifier model using pooling and dense layers."""
+
+import copy
+
+from keras import layers
+from tensorflow import keras
+
+from keras_cv.models.backbones.backbone_presets import backbone_presets
+from keras_cv.models.backbones.backbone_presets import (
+    backbone_presets_with_weights,
+)
+from keras_cv.models.classification.image_classifier_presets import (
+    classifier_presets,
+)
+from keras_cv.models.task import Task
+from keras_cv.utils.python_utils import classproperty
+
+
+@keras.utils.register_keras_serializable(package="keras_cv.models")
+class PooledDenseHead(layers.Layer):
+    """General purpose head for image classification.
+
+    Applies GlobalAveragePooling followed by a Dense layer mapping to the
+    number of classes to the Backbone output.
+
+    Args:
+        num_classes: int, number of classes to predict.
+        activation: A `str` or callable. The activation function to
+            use on the Dense layer. Set `classifier_activation=None` to return
+            the logits of the "top" layer.
+    """
+
+    def __init__(
+        self,
+        num_classes=2,
+        activation=None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.pooling = layers.GlobalAveragePooling2D(name="avg_pool")
+        self.dense = layers.Dense(
+            num_classes,
+            activation=activation,
+            name="predictions",
+        )
+        self.num_classes = num_classes
+        self.activation = activation
+
+    def call(self, inputs):
+        x = self.pooling(inputs)
+        return self.dense(x)
+
+    def get_config(self):
+        config = super().get_config()
+        config.update(
+            {
+                "num_classes": self.num_classes,
+                "activation": self.activation,
+            }
+        )
+        return config
+
+
+@keras.utils.register_keras_serializable(package="keras_cv.models")
+class ImageClassifier(Task):
+    """Image classifier with configurable prediction head.
+
+    Args:
+        backbone: `keras_cv.models.Backbone`, the backbone architecture of the
```

Comment:
What is special about a "backbone" ? Why not just a Model ? Any fwd pass will work here.